src/impl/regexParser.js

```diff
@@ -189,6 +190,34 @@ function extractASCII(match) {
   return [result, FixedOffsetZone.utcInstance];
 }
 
+const sqlDateRegex = /^(\d{4})-(\d\d)-(\d\d)$/,
+  sqlTimeRegex = /^(\d|\d\d):(\d\d):(\d\d)\.(\d\d\d)$/,
+  sqlDateTimeRegex = /^(\d{4})-(\d\d)-(\d\d) (\d|\d\d):(\d\d):(\d\d)\.(\d\d\d)$/;
```

Comment:
I strongly suspect there can be an arbitrary number of subsecond digits. I just accepted a fix for that in the ISO analog; you can probably extract the pad, parse, and rounding parts from there and put them in a `parseSecondFraction` function and call that from the handlers. Then just change `\d\d\d` to `\d{1,9}`

Reply:
you mean fraction second can be `12300000` and i have to trim the `0` right pad to get it?

Reply:
Well, more importantly it could be `234567890`, which we should parse as `235`. It's the equivalent of these tests: https://github.com/moment/luxon/blob/master/test/datetime/regexParse.test.js#L259